### ISSUE TITLE
Specialization of data selection

### ISSUE THREAD

**Issue by luispscarvalho** [DavidPinho/visminer-dashboard#4]
As an attempt to mitigate the overhead of data selection it is important to revise/refactor the data used to feed visualizations and/or graphical components of the frontend. As a potential (good) side-effect of data reduction/improvement we will possibly promote the readability of the source-code as a less amount of instructions will be necessary to manipulate datasets pulled from the servlet.

In short, this issue can be solved by:
1. Reducing the use of large json documents, so that they will contain only the essencial elements to make the visualization of information possible;
2. Refactoring the navigation in a way that document retrievals are performed progressively (by demand), which may possibly contribute for the division/reduction of documents.
